server/src/main/java/com/cloud/deploy/DeploymentPlanningManagerImpl.java

```diff
@@ -1448,10 +1452,15 @@ public int compare(Volume v1, Volume v2) {
                                 else
                                     requestVolumes = new ArrayList<Volume>();
                                 requestVolumes.add(vol);
-
+                                List<Pair<Volume, DiskProfile>> volumeDiskProfilePair = new ArrayList<>();
+                                for (Volume volume: requestVolumes) {
+                                    DiskOfferingVO diskOffering = _diskOfferingDao.findById(volume.getDiskOfferingId());
+                                    DiskProfile diskProfile = new DiskProfile(volume, diskOffering, _volsDao.getHypervisorType(volume.getId()));
+                                    volumeDiskProfilePair.add(new Pair<>(volume, diskProfile));
```

Comment:
possible to move creation of volume disk profile pair (from the volumes), to a common method and use it across?

Reply:
moved to a new method getVolumeDiskProfilePairs()